open-metadata-implementation/access-services/data-engine/data-engine-server/src/main/java/org/odpi/openmetadata/accessservices/dataengine/server/converters/ConnectorTypeConverter.java

```diff
@@ -0,0 +1,100 @@
+/* SPDX-License-Identifier: Apache 2.0 */
+/* Copyright Contributors to the ODPi Egeria project. */
+package org.odpi.openmetadata.accessservices.dataengine.server.converters;
+
+import org.odpi.openmetadata.accessservices.dataengine.model.ConnectorType;
+import org.odpi.openmetadata.commonservices.generichandlers.OpenMetadataAPIGenericConverter;
+import org.odpi.openmetadata.frameworks.connectors.ffdc.PropertyServerException;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.EntityDetail;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.InstanceProperties;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.Relationship;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.typedefs.TypeDefCategory;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.repositoryconnector.OMRSRepositoryHelper;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * ConnectorTypeConverter transfers the relevant properties from an Open Metadata Repository Services (OMRS)
+ * EntityDetail object into a ConnectorTypeElement bean.
+ */
+public class ConnectorTypeConverter<B> extends OpenMetadataAPIGenericConverter<B> {
+
+    public ConnectorTypeConverter(OMRSRepositoryHelper repositoryHelper, String serviceName, String serverName) {
+        super(repositoryHelper, serviceName, serverName);
+    }
+
+
+    /**
+     * Using the supplied instances, return a new instance of the bean. This is used for beans that have
+     * contain a combination of the properties from an entity and a that os a connected relationship.
+     *
+     * @param beanClass name of the class to create
+     * @param entity entity containing the properties
+     * @param methodName calling method
+     * @return bean populated with properties from the instances supplied
+     * @throws PropertyServerException there is a problem instantiating the bean
+     */
+    @Override
+    public B getNewBean(Class<B> beanClass, EntityDetail entity, String methodName) throws PropertyServerException {
+        try {
+            B returnBean = beanClass.getDeclaredConstructor().newInstance();
+
+            if (returnBean instanceof ConnectorType) {
+                ConnectorType bean = (ConnectorType) returnBean;
+
+                if (entity != null) {
+
+                    InstanceProperties instanceProperties = new InstanceProperties(entity.getProperties());
+
+                    bean.setQualifiedName(this.removeQualifiedName(instanceProperties));
+                    bean.setDisplayName(this.removeDisplayName(instanceProperties));
```

Comment:
I notice that you are removing properties from the instanceProperties in the entity. This is modifying a method parameter, I assume if the caller then addresses the entity it will no longer have all its instance properties. It seems a bit dangerous to have this side effect. It would be safer to clone the instance properties to a new object and remove properties from that object which only has this method scope.

Reply:
I was under the impression that the constructor of InstanceProperties does a deep copy and it's safe to use. This happens on line 47. Am I missing something?

Reply:
@davidradl This pattern is inherited from OpenMetadataAPIGenericConverter and it's widely used inside a series of converters. The purpose of it is to remove all properties from the deep copy that can be assigned to proper model members and have all the others inside the extendedProperties map.